src/main/java/core/basesyntax/dao/FruitDaoTransaction.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public interface FruitDaoTransaction {
```

Comment:
if this class works with your storage for read or write operations then it shouldn't do any other kind of work. Let's split your current logic. For transaction processing please create a separate service, for example, FruitTransactionProcessor. And FruitDaoTransaction should be renamed to FruitDao